test/utils/chrome.js

```diff
@@ -0,0 +1,265 @@
+    return this.win32()
+  }
+
+  win32() {
+    const installations = []
+    const sep = path.sep
+    const suffixes = [
+      `${sep}Chromium${sep}Application${sep}chrome.exe`,
+      `${sep}Google${sep}Chrome SxS${sep}Application${sep}chrome.exe`,
+      `${sep}Google${sep}Chrome${sep}Application${sep}chrome.exe`,
+      `${sep}chrome-win32${sep}chrome.exe`,
+      `${sep}Google${sep}Chrome Beta${sep}Application${sep}chrome.exe`
+    ]
+    const prefixes = [
+      process.env.LOCALAPPDATA,
+      process.env.PROGRAMFILES,
+      process.env['PROGRAMFILES(X86)']
+    ].filter(Boolean)
+    const customChromePath = this.resolveChromePath()
+    if (customChromePath) {
+      installations.push(customChromePath)
+    }
+    prefixes.forEach(prefix =>
+      suffixes.forEach((suffix) => {
+        const chromePath = path.join(prefix, suffix)
+        if (this.canAccess(chromePath)) {
+          installations.push(chromePath)
+        }
+      })
+    )
+    return installations
+  }
+
+  resolveChromePath() {
+    if (this.canAccess(process.env.CHROME_PATH)) {
+      return process.env.CHROME_PATH
+    }
+    if (this.canAccess(process.env.LIGHTHOUSE_CHROMIUM_PATH)) {
+      consola.warn(
+        'ChromeLauncher',
+        'LIGHTHOUSE_CHROMIUM_PATH is deprecated, use CHROME_PATH env variable instead.'
+      )
+      return process.env.LIGHTHOUSE_CHROMIUM_PATH
+    }
+  }
+
+  getLocalAppDataPath(path) {
+    const userRegExp = /\/mnt\/([a-z])\/Users\/([^/:]+)\/AppData\//
+    const results = userRegExp.exec(path) || []
+    return `/mnt/${results[1]}/Users/${results[2]}/AppData/Local`
+  }
+
+  sort(installations, priorities) {
+    const defaultPriority = 10
+    return installations
+      .map((inst) => {
+        for (const pair of priorities) {
+          if (pair.regex.test(inst)) {
+            return { path: inst, weight: pair.weight }
+          }
+        }
+        return { path: inst, weight: defaultPriority }
+      })
+      .sort((a, b) => b.weight - a.weight)
+      .map(pair => pair.path)
+  }
+
+  canAccess(file) {
+    if (!file) {
+      return false
+    }
+    try {
+      fs.accessSync(file)
+      return true
+    } catch (e) {
+      return false
+    }
+  }
+
+  findChromeExecutables(folder) {
+    const argumentsRegex = /(^[^ ]+).*/ // Take everything up to the first space
+    const chromeExecRegex = '^Exec=/.*/(google-chrome|chrome|chromium)-.*'
+    const installations = []
+    if (this.canAccess(folder)) {
+      // Output of the grep & print looks like:
+      //    /opt/google/chrome/google-chrome --profile-directory
+      //    /home/user/Downloads/chrome-linux/chrome-wrapper %U
+      let execPaths
+      // Some systems do not support grep -R so fallback to -r.
+      // See https://github.com/GoogleChrome/chrome-launcher/issues/46 for more context.
+      try {
+        execPaths = execSync(
+          `grep -ER "${chromeExecRegex}" ${folder} | awk -F '=' '{print $2}'`
```

Comment:
@clarkdo Could you help me with this? I'm getting error when run it on Alpine (Node v10.15.3)
I got
```
grep: unrecognized option: R
...
TypeError: input.on is not a function
    at new Interface (readline.js:189:11)
    at Object.createInterface (readline.js:69:10)
    at Promise (/builds/project/node_modules/puppeteer/lib/Launcher.js:327:25)
    at new Promise (<anonymous>)
    at waitForWSEndpoint (/builds/project/node_modules/puppeteer/lib/Launcher.js:326:10)
    at Launcher.launch (/builds/project/node_modules/puppeteer/lib/Launcher.js:170:41)
```
I think it's because no error is thrown inside try catch..